src/vfs2.c

```diff
@@ -1682,15 +1682,16 @@ static struct wal_hdr next_wal_hdr(const struct entry *e)
 
 static struct wal_frame_hdr txn_frame_hdr(struct entry *e,
 					  struct cksums sums,
-					  struct vfs2_wal_frame frame)
+					  const dqlite_vfs_frame *frame,
+					  uint32_t commit)
 {
 	struct wal_frame_hdr fhdr;
 
-	BytePutBe32(frame.page_number, fhdr.page_number);
-	BytePutBe32(frame.commit, fhdr.commit);
+	BytePutBe32((uint32_t)frame->page_number, fhdr.page_number);
```

Comment:
maybe it makes sense to change the page number in the struct definition from `unsigned long` to `uint32_t` (if that does not require a lot of changes).

Reply:
I agree, but unfortunately `dqlite_vfs_frame` appears in `dqlite.h` so it would be a breaking change. We could switch to a new frame type that uses `uint32_t`, basically amending this commit to just remove the `commit` from `vfs2_wal_frame`.